kDriveFileProvider/FileProviderExtension+Actions.swift

```diff
@@ -134,17 +136,21 @@ extension FileProviderExtension {
         if accessingSecurityScopedResource {
             fileURL.stopAccessingSecurityScopedResource()
         }
-        let importedItem = FileProviderItem(
-            importedFileUrl: storageUrl,
-            identifier: importedDocumentIdentifier,
-            parentIdentifier: parentItemIdentifier
-        )
-        fileProviderState.setImportedDocument(importedItem, forKey: importedDocumentIdentifier)
 
-        backgroundUploadItem(importedItem)
+        guard let parentDirectoryId = parentItemIdentifier.toFileId() else {
+            fatalError("missing parentDirectoryId TODO sentry")
+        }
 
-        manager.signalEnumerator(for: parentItemIdentifier) { _ in
-            completionHandler(importedItem, nil)
+        let importItem = UploadFileProviderItem(uploadFileUUID: importedFileUUID,
+                                                parentDirectoryId: parentDirectoryId,
+                                                userId: driveFileManager.drive.userId,
+                                                driveId: driveFileManager.drive.id,
+                                                sourceUrl: storageUrl,
+                                                conflictOption: .version,
+                                                shouldRemoveAfterUpload: false, /* should be true actually ?*/
```

Comment:
discuss with @PhilippeWeidmann about `shouldRemoveAfterUpload` shouldn't it be _true_ for FileProvider ?